asset-transfer-basic/chaincode-javascript/test/assetTransfer.test.js

```diff
@@ -0,0 +1,251 @@
+                assert.fail('CreateAsset should have failed');
+            } catch(err) {
+                expect(err.name).to.equal('failed inserting key');
+            }
+        });
+
+        it('should return success on CreateAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            let ret = JSON.parse((await chaincodeStub.getState('asset1')).toString());
+            expect(ret.ID).to.equal('asset1');
+            expect(ret.Color).to.equal('blue');
+            expect(ret.Size).to.equal(5);
+            expect(ret.Owner).to.equal('Tomoko');
+            expect(ret.AppraisedValue).to.equal(300);
+        });
+    });
+
+    describe('Test ReadAsset', () => {
+        it('should return error on ReadAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            try {
+                await assetTransfer.ReadAsset(transactionContext, 'asset2');
+                assert.fail('ReadAsset should have failed');
+            } catch (err) {
+                expect(err.message).to.equal('The asset asset2 does not exist');
+            }
+        });
+
+        it('should return success on ReadAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            let ret = JSON.parse(await chaincodeStub.getState('asset1'));
+            expect(ret.ID).to.equal('asset1');
+        });
+    });
+
+    describe('Test UpdateAsset', () => {
+        it('should return error on UpdateAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            try {
+                await assetTransfer.UpdateAsset(transactionContext, 'asset2', 'orange', 10, 'Me', 500);
+                assert.fail('UpdateAsset should have failed');
+            } catch (err) {
+                expect(err.message).to.equal('The asset asset2 does not exist');
+            }
+        });
+
+        it('should return success on UpdateAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            await assetTransfer.UpdateAsset(transactionContext, 'asset1', 'orange', 10, 'Me', 500);
+            let ret = JSON.parse(await chaincodeStub.getState('asset1'));
+            expect(ret.ID).to.equal('asset1');
+            expect(ret.Color).to.equal('orange');
+            expect(ret.Size).to.equal(10);
+            expect(ret.Owner).to.equal('Me');
+            expect(ret.AppraisedValue).to.equal(500);
```

Comment:
Build the object and do a `.eql` deep equal on the object, rather than comparing each key

Reply:
Fixed as indicated